packages/datadog_flutter_plugin/android/src/main/kotlin/com/datadoghq/flutter/DatadogLogEventMapper.kt

```diff
@@ -0,0 +1,116 @@
+package com.datadoghq.flutter
+
+import android.os.Handler
+import android.os.Looper
+import com.datadog.android.Datadog
+import com.datadog.android.log.LogsConfiguration
+import com.datadog.android.log.model.LogEvent
+import io.flutter.plugin.common.MethodChannel
+import java.util.Collections
+import java.util.concurrent.ConcurrentHashMap
+import java.util.concurrent.CountDownLatch
+import java.util.concurrent.TimeUnit
+
+/**
+ * This is a helper class that that simplifies event mapping / scrubbing for Log events.
+ *
+ * Since it is possible for Flutter engines to shut down and be recreated, it is possible that the
+ * object used as the event mapper to be lost. This class is used as a static instance that will
+ * perform mapping and call the Flutter mapping methods using the last provided MethodChannel.
+ */
+internal class DatadogLogEventMapper {
+    private val channels: MutableSet<MethodChannel> = Collections.newSetFromMap(ConcurrentHashMap())
+
+    fun addChannel(channel: MethodChannel) {
+        channels.add(channel)
+    }
+
+    fun removeChannel(channel: MethodChannel) {
+        channels.remove(channel)
+    }
+
+    fun attachMapper(config: LogsConfiguration.Builder): LogsConfiguration.Builder {
+        config.setEventMapper { event -> mapLogEvent(event) }
+
+        return config
+    }
+
+    @Suppress("TooGenericExceptionCaught")
+    internal fun mapLogEvent(event: LogEvent): LogEvent? {
+        val jsonEvent = event.toJson().asFlutterMap()
+        var modifiedJson: Map<String, Any?>? = null
+
+        val latch = CountDownLatch(1)
+
+        val handler = Handler(Looper.getMainLooper())
+        val channel = channels.firstOrNull() ?: return event
```

Comment:
just fyi: set/map doesn't have by default a guarantee for the items/keys ordering, so `first` may return any item from the set, not necessary first added. Not sure if it matters for you.

Upd: nevermind, you already have a similar comment in the `DatadogRumPlugin` 👍

Reply:
Yeah, I'm not sure I love the compromise but since the every engine is only ever working of a single codebase, any still active channel should hit the same code.